src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Drink;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// Servlet that handles search for drinks
+@WebServlet("/search")
+public class SearchServlet extends HttpServlet {
```

Comment:
Thinking aloud... if we represented a Drink as below, would simplify the search logic and be more space efficient (fewer Drink objects created), at the cost of a more complex DataStore representation.

```
Drink {
  name
  Map<storeId, RatingInfo>
}

RatingInfo {
  numRatings
  totalRating
}
```

Reply:
This is actually the original data structure we considered. But ultimately we chose not to implement this due to the complexity of using an embedded entity in datastore. I think (not 100% sure) Deron had a plan of including this in the design doc.

Reply:
Must be why I was hit with major deja vu. Please work with Deron to add a note to the design doc.